lib/classes/time/time.dart

```diff
@@ -0,0 +1,79 @@
+import"lib/classes/diretor/diretor.dart";
+import"lib/classes/tecnico/tecnico.dart";
+import"lib/classes/atleta/atleta.dart";
+import"lib/classes/Texto/Texto.dart";
+
+class Time {
+  String _nome, _cnpj;
+  double _fundoMonetario;
+  List _atletas = <Atleta>[];
+  Diretor _diretor;
+  Tecnico _tecnico;
+
+  Time(
+    this._nome,
+    this._cnpj,
+    this._fundoMonetario,
+    this._diretor,
+    this._tecnico,
+  );
+
+  get nome => this._nome;
+  get cnpj => this._cnpj;
+  set fundoMonetario(double dinheiro) => this._fundoMonetario = dinheiro;
+  get fundoMonetario => this._fundoMonetario;
+  set diretor(Diretor diretor) => this._diretor = diretor;
+  get diretor => this._diretor;
+  set tecnico(Tecnico tecnico) => this._tecnico = tecnico;
+  get tecnico => this._tecnico;
+  set atletas(Atleta atleta) => this._atletas.add(atleta);
+
+   void contratarAtleta(Atleta atleta, double valor, String termos,
+      DateTime dtinicio, DateTime dtfinal) {
+    if (this._tecnico.testarAtleta(atleta.idade, double.parse(atleta.peso)) ==true) 
+    {
+      String tipo = "Adesao";
+      this._diretor.fazerContrato(atleta, valor, termos, dtinicio, dtfinal,tipo);
+      this.atletas = atleta;
+      this._fundoMonetario -= valor;
+      print(Texto.contratarA(atleta.nome));
+    } else {
+      print(Texto.recusarA(atleta.nome));
+    }
+  }
+
+  void venderAtleta(Atleta atleta, double valor, String termos,DateTime dtinicio,DateTime dtfinal) 
+  {
+    String tipo = "Recisão";
+    this._diretor.fazerContrato(atleta, valor, termos,dtinicio,dtfinal,tipo);
+    this._atletas.removeWhere((atletas) => atleta.cpf == atletas.cpf);
+    this._fundoMonetario += valor;
+    print(Texto.venderA(atleta.nome));
+  }
+
+  void pagarAtleta(Atleta atleta, double valor) {
+    fundoMonetario -= valor;
+    atleta.salario += valor;
+    print(Texto.pagarA(atleta.nome));
+  }
+
+  PlanejamentoTreino _obterTreino() {
+    return this._tecnico.treino;
+  }
```

Comment:
mais elegante assim =)

PlanejamentoTreino get _obterTreino => this._tecnico.treino;